xwiki-platform-core/xwiki-platform-ldap/xwiki-platform-ldap-authenticator/src/main/java/com/xpn/xwiki/plugin/ldap/XWikiLDAPUtils.java

```diff
@@ -1194,6 +1213,153 @@ protected void updateUserFromLDAP(XWikiDocument userProfile, List<XWikiLDAPSearc
     }
 
     /**
+     * Sync user avatar with LDAP
+     * 
+     * @param ldapUid value of the unique identifier for the user to update.
+     * @param userProfile the XWiki user profile document.
+     * @param context the XWiki context.
+     * @return true if avatar was updated, false otherwise.
+     * @throws XWikiException 
+     */
+    protected boolean updatePhotoFromLdap(String ldapUid, XWikiDocument userProfile, XWikiContext context) throws XWikiException
+    {
+        XWikiLDAPConfig config = XWikiLDAPConfig.getInstance();
+
+        BaseClass userClass = context.getWiki().getUserClass(context);
+        BaseObject userObj = userProfile.getXObject(userClass.getDocumentReference());
+
+        // Get current user avatar
+        String userAvatar = userObj.getStringValue("avatar");
+        XWikiAttachment currentPhoto = null;
+        if(userAvatar != null)
+        {
+            currentPhoto = userProfile.getAttachment(userAvatar);
+        }
+
+        // Get properties
+        String photoAttachmentName = config.getLDAPParam(XWikiLDAPConfig.PREF_LDAP_PHOTO_ATTACHMENT_NAME, "ldapPhoto", context);
+        String ldapPhotoAttribute = config.getLDAPParam(XWikiLDAPConfig.PREF_LDAP_PHOTO_ATTRIBUTE, "thumbnailPhoto", context);
+
+        // Proceed only if any of conditions are true:
+        // 1. User do not have avatar currently
+        // 2. User have avatar and avatar file name is equals to PREF_LDAP_PHOTO_ATTACHMENT_NAME
+        if (userAvatar == null || userAvatar.isEmpty() || photoAttachmentName.equals(FilenameUtils.getBaseName(userAvatar)) || currentPhoto == null) {
+            // Obtain photo from LDAP
+            byte[] ldapPhoto = null;
+            List<XWikiLDAPSearchAttribute> ldapAttributes = searchUserAttributesByUid(
+                ldapUid, new String[]{ ldapPhotoAttribute }
+            );
+            if (ldapAttributes != null) {
+                for (XWikiLDAPSearchAttribute attribute : ldapAttributes) {
```

Comment:
Since you explicitly indicated only one attribute name the result should end up in the first element of the list.

        if (ldapAttributes != null && ldapAttributes.size() > 0) {
            ldapPhoto = ldapAttributes.get(0).byteValue;
        }